### PR TITLE
got some of priority queue tests to pass

### PR DESCRIPTION
**Pull request by lilvina**

### CODE REVIEW

File: src/pQueue.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+export default class pQueue {
+  constructor() {
+    this.storage = []
+    this.top = 0
```
Comment (shakalee14) [inline src/pQueue.js]: what is this top doing? doesnt pQueue have weight too?

File: src/pQueue.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  enqueue(name, priority) {
+    this.storage[this.top++] = {name, priority}
```
Comment (shakalee14) [inline src/pQueue.js]: nice!

File: src/pQueue.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return this.storage
+  }
+
+  isEmpty() {
+    return this.top === 0
+  }
+
+  length() {
+    return this.top
```
Comment (shakalee14) [inline src/pQueue.js]: just wondering, why did you call it top?

File: spec/pQueue.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      const aQueue = new pQueue()
+
+      aQueue.enqueue('dude', 90)
+      aQueue.enqueue('cool', 20)
+      aQueue.enqueue('turd', 50)
+      expect(aQueue.enqueue('foo',70)).to.eql([{name: 'dude', priority: 90}, {name: 'cool',  priority: 20}, {name: 'turd', priority: 50}, {name: 'foo', priority: 70}])
+    })
+  })
+
+  // context('front()', () => {
```
Comment (shakalee14) [inline spec/pQueue.js]: is this a WIP? are you going to remove this commented code or use it in the future?

File: spec/pQueue.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  })
+
+  context('enqueue()', () => {
+    it('adds an element with priority to the back of the queue.', () => {
+      const aQueue = new pQueue()
+
+      aQueue.enqueue('dude', 90)
+      aQueue.enqueue('cool', 20)
+      aQueue.enqueue('turd', 50)
+      expect(aQueue.enqueue('foo',70)).to.eql([{name: 'dude', priority: 90}, {name: 'cool',  priority: 20}, {name: 'turd', priority: 50}, {name: 'foo', priority: 70}])
```
Comment (jrobcodes) [inline spec/pQueue.js]: Consider formatting this line to make it more readable (it scrolls off the screen in github if greater than 80 characters):
```js
expect(aQueue.enqueue( 'foo',70 )).to.eql([
  {name: 'dude', priority: 90}, 
  {name: 'cool',  priority: 20}, 
  {name: 'turd', priority: 50}, 
  {name: 'foo', priority: 70}
])
```
This makes it a little for a developer outside of your project to read and understand your code.

Looking at this test, I think there is a flaw with your `enqueue` implementation - enqueue should not return anything, and you are testing the return value of `enqueue`.  Consider the following approach that leverages the API of the `PriorityQueue` object in order to perform the test:
```js
const queue = new pQueue()

queue.enqueue( 'highest', 90 )
queue.enqueue( 'lowest', 20 )
queue.enqueue( 'medium', 50 )
queue.enqueue( 'high medium', 70 )

expect( queue.front() ).to.equal( 'highest' )
```

Finally, consider renaming the identifier for your `PriorityQueue` class from `pQueue` to `PriorityQueue` - it is standard practice to use capital camel case for type identifiers, and to use lower camel case for variable identifiers.

File: spec/pQueue.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  //     expect(() => pQueue.back().to.eql(null))
+  //   })
+  // })
+
+  context('isEmpty()', () => {
+    it('returns true if the queue is empty.', () => {
+      const aQueue = new pQueue()
+
+      expect(() => aQueue.isEmpty().to.eql(true))
```
Comment (jrobcodes) [inline spec/pQueue.js]: This is incorrect syntax.  Your test should be:
```js
expect( aQueue.isEmpty() ).to.be.true
```
The general format for a test is:
```js
expect( VALUE ).EXPECTATION
```
Where VALUE could be a return value from a function (like `isEmpty`), and EXPECTATION is the code that validates your expectation of that VALUE (`.to.be.true` or `.to.eql( true )`)

File: spec/pQueue.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    })
+
+    it('returns false if queue is false.', () => {
```
Comment (jrobcodes) [inline spec/pQueue.js]: The test description should clearly reveal the intention of the test.  This description is unclear to me - what does it mean for a queue to be false?  I recommend a description like:
```
returns false if the queue contains no elements
```
This clearly describes the purpose of the `isEmpty` function, and the conditions under which this test should be successful.

File: src/pQueue.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+export default class pQueue {
```
Comment (jrobcodes) [inline src/pQueue.js]: I left this note above, but I'll repeat here: consider renaming this class to `PriorityQueue`.  It is best practice to use capital camel case for type (class) names, and lower camel case for variable names.

File: src/pQueue.js (hunk 3 above)
Comment (jrobcodes) [reply src/pQueue.js]: I agree with @shakalee14 comment here; in the context of a `PriorityQueue`, the name `front` would more clearly reveal the intention of this variable.

File: src/pQueue.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  constructor() {
+    this.storage = []
+    this.top = 0
+  }
+
+  enqueue(name, priority) {
+    this.storage[this.top++] = {name, priority}
+    return this.storage
```
Comment (jrobcodes) [inline src/pQueue.js:11]: `enqueue` should not return anything.  As well, this is a particularly risky implementation as it exposes the internal storage structure to clients of the `PriorityQueue`.  Our intent in creating objects is to encapsulate or hide the internal workings of the object, and only allow changes through the publicly defined API (like `enqueue`, `dequeue`, `isEmpty`, etc.)

**Pull request merged**